cpp/src/io/parquet/writer_impl.cu

```diff
@@ -2564,6 +2571,24 @@ std::unique_ptr<std::vector<uint8_t>> writer::impl::close(
       }
     }
 
+    // set row group ordinals
+    auto iter        = thrust::make_counting_iterator(0);
+    auto& row_groups = fmd.row_groups;
+    std::for_each(
+      iter, iter + row_groups.size(), [&row_groups](auto idx) { row_groups[idx].ordinal = idx; });
+
+    // set sorting_columns on row groups
+    if (_sorting_columns.has_value()) {
+      auto const& sorting_cols = _sorting_columns.value();
+      std::for_each(iter, iter + row_groups.size(), [&row_groups, &sorting_cols](auto idx) {
```

Comment:
This seems expensive. Have we run any numbers on performance here? Would it make sense to push this to the GPU? If so, I'm perfectly fine with an investigation issue if this isn't a big hit.

Reply:
This should be insignificant.  A large file might have a dozen or so row groups, so this should be really fast. If there was a struct per page, it would be a different story.

Honestly, I don't know why this is defined per row group...it should be at the file level :(